hatch/src/yaml.rs

```diff
@@ -32,22 +40,86 @@ pub fn parse(file_path: &Path) -> HatchResult<Project> {
     return Err(MissingNameError)?;
   }
 
-  if let Some(b) = yml_vec[0]["build"].as_str() {
-    kind = match b {
-      "static-lib" => ProjectKind::Library(LibraryKind::Shared),
-      "shared-lib" => ProjectKind::Library(LibraryKind::Static),
-      _ => ProjectKind::Binary
-    }
-  } else {
-    return Err(MissingBuildError)?;
-  }
-
   if let Some(v) = yml_vec[0]["version"].as_str() {
     version = v.to_owned();
   } else {
     return Err(MissingVersionError)?;
   }
 
+  if let Some(configurations) = yml_vec[0]["build"].as_hash() {
+    let items = configurations
+      .iter()
+      .map(|(k, v)| {
+        (k.as_str(), v.as_str())
+      })
+      .filter(|&(k, v)| {
+        k.is_some() && v.is_some()
+      })
+      .map(|(k, v)| {
+        (k.unwrap(), v.unwrap())
+      }).collect::<Vec<(&str, &str)>>();
```

Comment:
`filter_map`?

Reply:
The issue is that I need to propagate the tuple, filter_map requires that a single Option is returned; therefore, I cannot use it for the tuple.